Social - The Social Media

Nastasha1888 opened the pull request.

Nastasha1888 committed:
first run

Nastasha1888 committed:
added ON DELETE CASCADE to notifications

Nastasha1888 committed:
changes OR to AND in event

Nastasha1888 committed:
added ON DELETE CASCADE to friends

Nastasha1888 committed:
added ssecond ON DELETE CASCADE to friends

Nastasha1888 committed:
added more cascades

Nastasha1888 committed:
added missing comma

Nastasha1888 committed:
removed extra comma....

halverneus commented on week9/answers.sql:
```diff
@@ -1,3 +1,106 @@
+
+CREATE TABLE users (
+  user_id INT UNSIGNED NOT NULL PRIMARY KEY AUTO_INCREMENT,
+  first_name VARCHAR(30),
```
I would like to see more "NOT NULL" assignments scattered through the fields. This provides a lot of protection for you database, especially as a developer making calls to your schema (don't want your code to crash when calculating with a TIMESTAMP that was set to NULL, or expecting a string for a first_name value and getting NULL back (and having your program crash with a NULL-dereference error).

halverneus commented on week9/answers.sql:
```diff
@@ -1,3 +1,106 @@
+    p.post_id,
+    p.content
+FROM notifications n
+JOIN users u ON u.user_id = (
```
JOIN -> INNER JOIN (multiple places)

No need for nested-SELECT statements in this assignment (adds significant performance and readability overhead in this case compared to simply "LEFT OUTER JOIN"). Keep it simple. LLMs don't always give the elegant answer, rather they tend to convert sentences literally.

halverneus commented on week9/answers.sql:
```diff
@@ -1,3 +1,106 @@
+
+DELIMITER ;;
+
+CREATE TRIGGER after_user_insert 
```
The intent of this exercise is to practice more with CURSORs rather than using the nested SELECT (not introduced in this class).

halverneus commented on week9/answers.sql:
```diff
@@ -1,3 +1,106 @@
+    WHERE u.user_id != NEW.user_id;
+END;;
+
+CREATE PROCEDURE add_post(IN user_id INT UNSIGNED, IN content VARCHAR(255))
```
Again, this is intended to be a CURSOR exercise.

halverneus commented on week9/answers.sql:
```diff
@@ -1,3 +1,106 @@
+DO
+    DELETE FROM sessions 
+    WHERE updated_on < DATE_SUB(NOW(), INTERVAL 2 HOUR)
+    AND created_on < DATE_SUB(NOW(), INTERVAL 2 HOUR);;
```
`created_on` can be ignored.

halverneus reviewed (commented):
Most of the reason for the low score on this assignment was due to using nested SELECT statements vs CURSORs (a warning comment I made in each class, as we didn't cover nested SELECT statements). Feel free to make any changes and resubmit, if you choose. If you are happy with your score, you can merge this Pull Request.

Nastasha1888 committed:
Added cursors

Nastasha1888 committed:
removed DELIMITER ;

Nastasha1888 committed:
moved declare

Nastasha1888 committed:
moved other declare

Nastasha1888 committed:
again fixing order

Nastasha1888 committed:
trying to fix ERROR 1452 (23000) at line 3:

Nastasha1888 committed:
same

Nastasha1888 committed:
Update answers.sql

Nastasha1888 committed:
Update answers.sql

Nastasha1888 committed:
Update answers.sql

Nastasha1888 committed:
trying to fix trigger and procedure

Nastasha1888 committed:
same

Nastasha1888 committed:
samme

Nastasha1888 committed:
same again

Nastasha1888 committed:
Update answers.sql

Nastasha1888 committed:
Update answers.sql

changed:
      INSERT INTO fetched_users 
        (current_post_id, current_user_id) 
      VALUES 
        (current_post_id, current_user_id);
to:
      INSERT INTO fetched_users 
        (post_id, user_id) 
      VALUES 
        (current_post_id, current_user_id);

Nastasha1888 committed:
moved fetch_users table outside of delimiter

Nastasha1888 committed:
Update answers.sql

Nastasha1888 committed:
fixed typo

Nastasha1888 committed:
Update answers.sql

Nastasha1888 committed:
Update answers.sql

Nastasha1888 committed:
Update answers.sql

Nastasha1888 committed:
fixed typo

Nastasha1888 committed:
fixed another typo..

Nastasha1888 committed:
Update answers.sql

halverneus commented on week9/answers.sql:
```diff
@@ -1,3 +1,162 @@
+    DECLARE friends_cursor CURSOR FOR 
+      SELECT f.friend_id 
+        FROM friends f
+        WHERE f.user_id = p_user_id
```
Only need the query up until here. The UNION and following statement is not required.

halverneus reviewed (commented):
Looking great! I would be more judicious about my use of NOT NULL for all columns that shouldn't have a NULL value. If a consumer of your database inserts NULL data, it could break many of your queries.

You probably know by now: change for a higher grade or merge if you are happy. Grade is in the gradebook, now.

Nastasha1888 committed:
added NOT NULL to foreign keys and removed UNION

Nastasha1888 committed:
added missing ;

halverneus reviewed (commented):
Gradebook has been updated. Have a great winter break!
